src/NUnitConsole/nunit-console/Utilities/TeamCityEventHandler.cs

```diff
@@ -45,10 +46,20 @@ public TeamCityEventHandler(TextWriter outWriter)
             _outWriter = outWriter;
         }
 
-        public void TestStarted(XmlNode node)
+        public void SuiteStarted(XmlNode testEvent)
         {
-            string name = node.GetAttribute("fullname");
-            string testId = node.GetAttribute("id");
+            _asemblyResolver.RegisterSuite(testEvent.GetAttribute("id"), testEvent.GetAttribute("fullname"));
+        }
+
+        public void SuiteFinished(XmlNode testEvent)
+        {
+            _asemblyResolver.UnregisterSuite(testEvent.GetAttribute("id"), testEvent.GetAttribute("fullname"));
+        }
+
+        public void TestStarted(XmlNode testEvent)
+        {
+            string testId = testEvent.GetAttribute("id");
+            string name = this.EnrichTestName(testId, testEvent.GetAttribute("fullname"));
```

Comment:
Can a user change the fullname attribute for the assembly suite?
We are receiving `<start-suite id="0-1038" name="HelloWorld.Tests.dll" fullname="SomePath\HelloWorld.Tests.dll" />` when assembly suite is starting. The xml data does not contain the classname attribute. I've tried to use current xml contract. Can I add the attribute classname for the assembly suite?